### ISSUE TITLE
[spec] Phase B -- optimizer control ablation + QK-gain ablation

### ISSUE THREAD

**Issue by gHashTag** [gHashTag/t27#1002]
## Goal

Implement Phase B of the phi->IGLA epic (#181): the referee-demanded control
ablation that puts phi-derived hyperparameters against equal-budget baselines.

Two spec changes:

1. **B1 (optimizer)**: Reconcile `specs/ml/optimizer/adamw.t27` to the
   canonical phi anchor (`beta1=phi^(-1)~0.618`, `weight_decay=phi^(-3)~0.236068`)
   and add a four-cell control-test design comparing phi variants against
   tuned-standard and random-rational controls.

2. **B2 (QK-Gain)**: Add a three-cell control-test to `specs/nn/attention.t27`
   comparing `gain=phi^2`, `gain=1.0` (standard), and learned gain.

All phi claims are labeled `[Open conjecture]`. Only `phi^2 + phi^(-2) = 3`
is `[Verified]` (Law L5).

## Why this matters

The existing spec (`adamw.t27`) uses `PHI_BETA1 = 0.9/phi ~ 0.556`, while the
trainer (`igla-peek/src/optimizer.rs`) uses `beta1 = phi^(-1) ~ 0.618`. These
are two different phi-parametrisations with no reconciliation and no
falsification test. Without a control, any phi claim is unfalsifiable --
which the honesty mandate forbids.

For QK-Gain: `igla-peek/src/attention.rs` INV-9 anchors `qk_gain=phi^2` with
language suggesting it is "mathematically-optimized". phi^2 = phi + 1 is
[Verified arithmetic] (Law L5), but that phi^2 is the *optimal* attention
gain has no theoretical basis. A control test is required.

## Source of truth

- `specs/ml/optimizer/adamw.t27` -- B1 target
- `specs/nn/attention.t27` -- B2 target
- `igla-peek/src/optimizer.rs` -- canonical phi anchor reference
- `igla-peek/src/attention.rs` -- INV-9 qk_gain reference

### B1: `specs/ml/optimizer/adamw.t27`

New constants:
```
PHI_INV : gf16::GF16 = 0.618033988749895   // phi^(-1)  [Open conjecture]
PHI_INV_CUBED : gf16::GF16 = 0.236067977499790  // phi^(-3)  [Open conjecture]
// Note: brief had 0.118034 but phi^(-3) = 0.236068 (trainer test confirms)
PHI_CANONICAL_BETA1 = PHI_INV
PHI_CANONICAL_WEIGHT_DECAY = PHI_INV_CUBED
PHI_DAMPED_BETA1 = 0.9 / PHI               // old PHI_BETA1, renamed
RATIONAL_BETA1 : gf16::GF16 = 0.5714285714285714  // 4/7, non-Fibonacci control
RATIONAL_WEIGHT_DECAY : gf16::GF16 = 0.2333333333333333 // 7/30, comparable cost
```

New `PhiVariant` enum: `Canonical | Damped | TunedStd | RandomRat`

New function: `get_effective_betas(variant: PhiVariant) -> (b1, b2, wd)`

New test blocks (B1-CONST-1..6, B1 variant configs, ablation test)  
New invariant blocks (B1-INV-1..7)  
New bench block (`b1_ablation_equal_budget_throughput`)

### B2: `specs/nn/attention.t27`

New constants:
```
GAIN_PHI_SQ : gf16::GF16 = 2.618033988749895  // phi^2  [Open conjecture]
GAIN_STANDARD : gf16::GF16 = 1.0
GAIN_SQRT_PHI : gf16::GF16 = 1.272019649514069
```

New `QKGainVariant` enum: `PhiSquared | Standard | Learned | SqrtPhi`  
New `QKGainConfig` struct  
New functions: `get_qk_gain()`, `effective_score_scale()`  
New test blocks (B2-CONST-1..5, B2 variant tests, ablation test)  
New invariant blocks (B2-INV-1..6)  
New bench blocks (B2 forward pass per variant)

## Done when

- `./scripts/tri seal specs/ml/optimizer/adamw.t27 --save` passes (seal updated)
- `./scripts/tri seal specs/nn/attention.t27 --save` passes
- `./scripts/tri test` passes with all new test/invariant blocks
- `./scripts/tri validate-conformance` passes (no FORMAT-SPEC-001 changes)
- `./scripts/tri validate-gen-headers` passes
- All phi claims in new blocks are labeled `[Open conjecture]`
- No hype words (breakthrough/Nobel/proves/revolutionary) in any new text
- `docs/NOW.md` updated with today's date (now-sync-gate)

## How to verify

```bash
# In t27 repo root after applying the spec changes:

# 1. Seal the modified specs
./scripts/tri seal specs/ml/optimizer/adamw.t27 --save
./scripts/tri seal specs/nn/attention.t27 --save

# 2. Run all tests
./scripts/tri test

# 3. Validate conformance (must not touch FORMAT-SPEC-001)
./scripts/tri validate-conformance

# 4. Validate generated headers (gen/ is generated, not hand-edited)
./scripts/tri validate-gen-headers

# 5. Check now-sync-gate
./scripts/tri check-now

# 6. Verify new constants numerically (quick sanity check):
# PHI_CANONICAL_BETA1 = 1/phi ~ 0.618034
# PHI_CANONICAL_WEIGHT_DECAY = phi^(-3) ~ 0.236068
# PHI_DAMPED_BETA1 = 0.9/phi ~ 0.55557
# PHI_SQ = phi + 1 ~ 2.618034
# All values must be ASCII-representable floats in the spec file.
```

## Risks

1. **Breaking existing tests** that depend on `PHI_BETA1` name or the
   `use_phi_betas : bool` field. Mitigation: rename carefully; check all
   callers in `gen/` and `tests/` before posting PR.

2. **FORMAT-SPEC-001.json** (L6 ceiling) must not be changed. The new
   constants use `gf16::GF16` which is already registered. No new format types.

3. **Now-sync-gate**: `docs/NOW.md` and root `NOW.md` must carry today's date
   or CI will reject the PR.

4. **Falsification outcome**: If B1 runs show that tuned-standard or
   random-rational AdamW matches phi-canonical within error, the parent
   agent must update the claim labels from `[Open conjecture]` to
   `[Falsified for this regime]` in a follow-on issue.

5. **muP interaction**: If the project adopts muP (Yang 2022), the LR
   scaling rules change (phi-independent). The B1 ablation should be re-run
   at the muP-transferred LR for a fair comparison.

## Links

- Closes #181 (phi->IGLA epic, parent issue)
- igla-peek/src/optimizer.rs (canonical phi anchor source)
- igla-peek/src/attention.rs (INV-9 qk_gain source)
- arXiv:2404.04454 Xie & Li 2024 (AdamW L-inf bias baseline)
- arXiv:2405.13698 Wang & Aitchison 2025 (weight-decay scaling baseline)
- arXiv:2510.12402 Chen 2025 (Cautious Weight Decay baseline)
- arXiv:1902.07638 Li & Talwalkar 2019 (random-search baseline)
- arXiv:1707.05589 Melis/Dyer/Blunsom 2017 (equal-budget protocol)
- arXiv:2203.03466 Yang et al. 2022 (muP, phi-independent LR transfer)
- arXiv:2203.15556 Hoffmann 2022 (BPB/Chinchilla, metric anchor)
